designs/0005-static-matrices.md

```diff
@@ -0,0 +1,240 @@
+- Feature Name: static_matrices
+- Start Date: April 30th, 2020
+- RFC PR: [Example Branch](https://github.com/stan-dev/math/compare/feature/var-template)
+- Stan Issue: [#1805](https://github.com/stan-dev/math/issues/1805)
+
+# Summary
+[summary]: #summary
+
+This proposes a generalization of the reverse mode autodiff type in Stan Math,
+`var_value<T>`, where `T` is the underlying storage type of the autodiff variable.
+For `T == double`, this type is equivalent to the current Stan Math `var` type.
+
+The usefulness of this type comes from setting `T` to more general data structures,
+in particular the Eigen Matrix types. Instead of dealing with matrices of vars,
```

Comment:
Suggestion:

1. Move the "The usefulness..." sentence to the paragraph above and add a paragraph break after it. It'd be nice to have the usefulness of the proposed generalization right along with the introductory paragraph.

2. This allows the second paragraph to focus on the point about matrix of vars vs var_value of matrix. Suggested replacement to make it clearer (please feel free to ditch, change, etc.): "Instead of dealing with matrices of `var` (`Eigen::Matrix<var>`), this allows operations in the Math library to be written in terms of a generalized var of matrices (`var_value<Eigen::Matrix<double>>`)." And then continue with the next sentence.